### PR TITLE
CORE-5919 - RegistrationState is null during verification

### PR DESCRIPTION
**Pull request by nikinagy**
https://r3-cev.atlassian.net/browse/CORE-5919

### CODE REVIEW

**Review by charlieR3 — DISMISSED** [review]
Looks good but build failed.

File: components/membership/membership-p2p-impl/src/main/kotlin/net/corda/membership/impl/p2p/handler/RegistrationRequestHandler.kt [hunk 1]
```diff
@@ -23,14 +24,16 @@ internal class RegistrationRequestHandler(
         payload: ByteBuffer
     ): Record<String, RegistrationCommand> {
         logger.info("Received registration request. Issuing StartRegistration command.")
+        val registrationRequest = avroSchemaRegistry.deserialize<MembershipRegistrationRequest>(payload)
+        val registrationId = registrationRequest.registrationId
         return Record(
             REGISTRATION_COMMAND_TOPIC,
-            header.source.toCorda().shortHash,
+            registrationId + "-" + header.destination.toCorda().shortHash,
```
Comment (YashNabar) [inline components/membership/membership-p2p-impl/src/main/kotlin/net/corda/membership/impl/p2p/handler/RegistrationRequestHandler.kt]: Can we use interpolation instead?

**Review by YashNabar — APPROVED** [review]
LGTM

File: components/membership/membership-p2p-impl/src/main/kotlin/net/corda/membership/impl/p2p/handler/RegistrationRequestHandler.kt (hunk 1 above)
Comment (nikinagy) [reply components/membership/membership-p2p-impl/src/main/kotlin/net/corda/membership/impl/p2p/handler/RegistrationRequestHandler.kt]: updated

**Review by charlieR3 — APPROVED** [review]

**Pull request merged**